src/scripting.cpp

```diff
@@ -239,17 +239,17 @@ QJSValue Scripting::position(int x, int y) {
 }
 
 Tile Scripting::toTile(QJSValue obj) {
-    if (!obj.hasProperty("tileId")
-     || !obj.hasProperty("xflip")
-     || !obj.hasProperty("yflip")
-     || !obj.hasProperty("palette")) {
-        return Tile();
-    }
     Tile tile = Tile();
-    tile.tileId = obj.property("tileId").toInt();
-    tile.xflip = obj.property("xflip").toBool();
-    tile.yflip = obj.property("yflip").toBool();
-    tile.palette = obj.property("palette").toInt();
+
+    if (obj.hasProperty("tileId"))
```

Comment:
double checking... are these calls to `.toInt()` and `.toBool()` still valid/functional after their type changes in `tile.h`?

Reply:
Yeah it's converting the JS property string to int/bool, and then converting (implicitly) to the bitfield types. `bool` to `uint16_t:1` is fine, `int` to `uint16_t:4` etc. might give a warning about dropping higher bits but that's what we want here anyway